expression/builtin_time.go

```diff
@@ -1835,7 +1835,55 @@ type builtinPeriodAddSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_period-add
 func (b *builtinPeriodAddSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("PERIOD_ADD")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if args[0].IsNull() || args[1].IsNull() {
+		return d, errors.Trace(err)
+	}
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+	period, err := args[0].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	//Check zero
+	if period <= 0 {
+		d.SetNull()
+		return d, errors.Errorf("in the format YYMM or YYYYMM")
+	}
+
+	y := period / 100
+	m := period % 100
+
+	// YYMM, 00-69 year: 2000-2069
+	// YYMM, 70-99 year: 1970-1999
+	// YYYYMM, 1000-? year: 1000-?
+	if y <= 69 {
+		y += 2000
+	} else if y <= 99 {
+		y += 1900
+	} else if y < 1000 {
```

Comment:
y < 1000 is also valid 
mysql> select period_add(12323,10);
+----------------------+
| period_add(12323,10) |
+----------------------+
|                12509 |
+----------------------+
1 row in set (0.00 sec)